Sources/CacheStore/Stores/CacheStore/ScopedKeyValueCacheStore.swift

```diff
@@ -0,0 +1,38 @@
+//
+//  ScopedKeyValueCacheStore.swift
+//  
+//
+//  Created by Leif on 11/2/22.
+//
+
+class ScopedKeyValueCacheStore<Key: Hashable, Value, ScopedKey: Hashable, ScopedValue>: CacheStore<ScopedKey> {
+    weak var parentCacheStore: CacheStore<Key>?
+    private var keyValueTransformation: BiDirectionalTransformation<(Key, Value?)?, (ScopedKey, ScopedValue?)?>
+
+    init(
+        keyValueTransformation: BiDirectionalTransformation<(Key, Value?)?, (ScopedKey, ScopedValue?)?>
+    ) {
+        self.keyValueTransformation = keyValueTransformation
+
+        super.init(initialValues: [:])
+    }
+
+    required init(initialValues: [ScopedKey: Any]) { fatalError("Not implemented") }
```

Comment:
Should we ever `fatalError` something, especially marked `required`?

Reply:
Great question! Since this is an `internal` class we are the only ones that can call this. It is required because this class extends from CacheStore and that is its init.. But we want to use a new idea of a transformation to create a CacheStore from another one.

We could change this.. but then we need to init and need a `BiDirectionalTransformation<(Key, Value?)?, (ScopedKey, ScopedValue?)?>` to do so.. FatalError allows us to just not use this internally.